### PR TITLE
[Bug] Enable font overwrites to be passed through chart template

### PR DESCRIPTION
**Pull request by huong-li-nguyen**
## Description

Below previously did not work, because we defined a font family for subcategories of the charts e.g. xaxis_title_font, xaxis_tickfont, etc. However, if we only define the `font_family` in our template
it will be passed on to the chart's subcomponents and this way, users only have to change one argument
if they want to change the font globally, instead of overwriting the font for every single subcomponent.

```
pio.templates["vizro_dark"]["layout"]["font_family"] = "PlayfairDisplay, Inter, sans-serif, Arial, serif"
pio.templates["vizro_light"]["layout"]["font_family"] = "PlayfairDisplay, Inter, sans-serif, Arial, serif"
```

Will be merged before https://github.com/mckinsey/vizro/pull/716

## Screenshot
See scratch_dev - this works now:

Screenshot 2024-09-18 at 18 51 33

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-core/examples/scratch_dev/assets/css/custom.css [hunk 1]
```diff
@@ -1,3 +1,8 @@
-#page-header {
-  padding-left: 8px;
+@font-face {
```
Comment (huong-li-nguyen) [inline vizro-core/examples/scratch_dev/assets/css/custom.css:1]: I'll remove the custom font before merge, this is just for testing

**Review by petar-qb — APPROVED** [review]
I like it! 🏅 

I tested it with (`app.py`):
```py 
pio.templates["vizro_dark"]["layout"]["font_family"] = "Courier New"
pio.templates["vizro_light"]["layout"]["font_family"] = "Brush Script MT"
```

and only this (`custom.css`):
```css
* {
  font-family: Courier New;
}
```

and it works perfectly 🎉 

---

Should we add it as an example somewhere in the docs?

**Comment by huong-li-nguyen** [comment]
> Should we add it as an example somewhere in the docs? WDYT @huong-li-nguyen

It's here: https://github.com/mckinsey/vizro/pull/716 :) I was writing this up, when I noticed the bug 👍 I'll merge this PR before the other one

**Comment by petar-qb** [comment]
> It's here: https://github.com/mckinsey/vizro/pull/716 :) I was writing this up, when I noticed the bug 👍 I'll merge this PR before the other one

Awesome, it's ready to go by my opinion then 😃
May you just reference the 716 in the description of this PR?

**Review by antonymilne — APPROVED** [review]
This is much better now 👍

**Comment by huong-li-nguyen** [comment]
@l0uden - just for my own tracking - I'll merge this PR even though Vizro QA tests trigger fails. It seems to be a false positive that has been failing on previous PRs and main already. Would be good if we can double-check this:

Screenshot 2024-09-19 at 11 38 00

**Pull request merged**